### ISSUE TITLE
Building ocp cluser in vmware with static ip using ansible. 

### ISSUE THREAD

**Issue by kiransharma2016** [sa-ne/openshift4-vmware-upi#3]
Did pusihng network info through the playbook work? we are using openshift 4.5 and  passing the network does not work though I can have name attribute. Although I was able to pass the ignition /append igniton file.

**Comment by nasx** [comment]
@kiransharma2016 using static IPs (i.e. setting `guestinfo.afterburn.initrd.network-kargs`) requires OpenShift 4.6.

Separately for 4.6, there were some issues with the templating I was using. I just merged the updates.

**Comment by nasx** [comment]
@kiransharma2016 also just remembered, this repository only works with 4.6 out of the box. With 4.6, the ignition version changed from v2 to v3. If you want to make it work w/ OpenShift 4.5, you need to modify the template `roles/vmware/templates/append-bootstrap.j2` and change the version from 3.1.0 to 2.1.0. You can see the complete changes in the link below:

https://github.com/sa-ne/openshift4-vmware-upi/commit/af41cc45fd5c9636d210aca272f56cef76c13fce#diff-4ec6eea3d93c64b8c08506ab5162455425a56b426d235f7391c584e1754d1313L1-R12

**Comment by kiransharma2016** [comment]
@nasx 
yes, I read about this.  Couple of question

1. Are you saying  v4.6 accepts "guestinfo.afterburn.initrd.network-kargs" parameters? or We also tried different ignition files
{
  "ignition": { "version": "2.0.0" },
  "networkd": {
    "units": [
      {
        "name": "00-eth.network",
        "contents": "[Match]\nName=eth*\n\n[Network]\nBond=bond0"
      },
      {
        "name": "10-bond0.netdev",
        "contents": "[NetDev]\nName=bond0\nKind=bond"
      },
      {
        "name": "20-bond0.network",
        "contents": "[Match]\nName=bond0\n\n[Network]\nDHCP=true"
      }
    ]
  }
Ref: https://coreos.com/ignition/docs/latest/network-configuration.html

It takes certain parameters eg ifconfig file but does not take the ip so we stil have to enter ip in boot up process. 

With v4.6 are you able to setup ip and other network information ? no manual process involved? 

I do not think the vmware guest_module is changed to access network attribute except name.

**Comment by romankspb** [comment]
@kiransharma2016 Look at this https://github.com/romankspb/okd4-cluster
May be it help you.

**Comment by nasx** [comment]
@kiransharma2016 I believe you can set the network info via ignition, but you will also need to set this information via kernel parameters on boot as well.

With 4.6 and VMware, you can specify `guestinfo.afterburn.initrd.network-kargs` as part of the VMs vApp properties (similar to how you define ignition location). This method requires no manual intervention. This is done in the `vmware` role here:

https://github.com/sa-ne/openshift4-vmware-upi/blob/master/roles/vmware/tasks/vms-cluster-placement.yaml#L27:L46

**Comment by kiransharma2016** [comment]
We are not planning to move to 4.6 so I can test my automation now. 

I got my ansbile playbook working, it creates a vm but no netwok. 

I used the following in the vm_guest module:

 vapp_properties:
      - key: "guestinfo.ignition.config.data"
        value: "{{ lookup('template', 'templates/worker.j2') | to_json | b64encode }}"
      - key: "guestinfo.ignition.config.data.encoding"
        value: base64
      - key: "disk.EnableUUID"
        value: "TRUE"
      - key: 'guestinfo.afterburn.initrd.network-kargs',
        value: ip_string

Instead of setting facts, ( I treid that also but no change ) and using that in customization, I used all parameters in vapp_properties stanza as above. Also,I ahve to use "id"  instead of "key". Syntactically it does not complain and creates vm but without network

I thought we need to have some sort of wating so I added another task to wait for 5 min. STill invein. 

Are you able to crete vm and join the cluster? 

Note: I  have ip_string as variable above
